### PR TITLE
[v1.2][ISSUE-306]Support text_prompt operator

### PR DESCRIPTION
**Pull request by haojinIntel**
### What changes were proposed in this pull request?

Support prompt for various datasets.

### Why are the changes needed?

Enhance RecDP

### How was this patch tested?

UT

### CODE REVIEW

**Comment by xuechendi** [comment]
@haojinIntel , CICD reports unfound package. Do we need to specify the version?
image

**Comment by haojinIntel** [comment]
> @haojinIntel , CICD reports unfound package. Do we need to specify the version? image

@xuechendi  Hi chendi. The failed reason is that promptsource does not support working on python-3.10.

**Comment by xuechendi** [comment]
@haojinIntel , update setup.py using pip install from github

**Comment by xuechendi** [comment]
@haojinIntel , seems reader still show some issues

**Comment by haojinIntel** [comment]
image
@xuechendi Hi, chendi. The failed unit test is not related with JsonReader.

**Comment by haojinIntel** [comment]
@yao531441 Hi Qing, Please fix the unit test issue about url_filter.

File: RecDP/pyrecdp/primitives/operations/text_reader.py [hunk 1]
```diff
@@ -23,28 +23,29 @@ def __init__(self, input_dir = ""):
...
-        def convert_json(s):
-            if isinstance(s, str):
-                content = json.loads(s)
-            elif isinstance(s, dict):
-                content = json.loads(s['text'])
-            if 'meta' in content:
-                content['meta'] = str(content['meta'])
-            return content
-        self.cache = rd.read_text(self.input_dir).map(convert_json)
+        self.cache = rd.read_json(self.input_dir)
```
Comment (xuechendi) [inline RecDP/pyrecdp/primitives/operations/text_reader.py:26]: @haojinIntel , maybe we should also use some mixed schema jsonl to test? I had encountered with ray read error when jsonl contains nested scheme, that is why I did some manual fix in above

**Review by xuechendi — APPROVED** [review]
Codes looks good!  reader test with broken jsonl files can be done in future PR

**Pull request merged**